src/DynamoCoreWpf/ViewModels/RunSettingsViewModel.cs

```diff
@@ -236,6 +236,14 @@ public RunSettingsViewModel(RunSettings settings, HomeWorkspaceViewModel workspa
             ToggleRunTypeEnabled(RunType.Periodic, false);
         }
 
+        /// <summary>
+        /// When switching workspace, this need to be called in HomeworkspaceViewModel dispose function
+        /// </summary>
+        internal void Dispose()
+        {
+            this.workspaceViewModel = null;
```

Comment:
I think the cases where this needs to be done in c# are small in number, why do we need to do it here?

Reply:
I believe something here is preventing HomeworkspaceViewModel to be disposed, not sure if it is related to the readonly field or simply calling dispose, will give it another try tomorrow

Reply:
From the retention paths you captured screenshots of it looks like the `DynamoView` is indirectly holding onto an instance of `RunSettingsViewModel` that is in turn holding onto the `HomeWorkspaceViewModel`. I think we should try and dispose off the `HomeWorkspaceViewModel` from `DynamoView` in that case, no?

Reply:
I believe DynamoView is a singleton as well as the RunSettingsControl. So we won't be able to know when to dispose specifically from the view, but we can do that when closing the currentWorkspace. WorkspaceRemoved event will be triggered and then Dispose() on HomeWorkspaceModel will be called, that's why I added the logic there.